lib/field.mli

```diff
@@ -0,0 +1,56 @@
+type phrase = Rfc5322.phrase
+type phrase_or_message_id = Rfc5322.phrase_or_message_id
+
+type 'a t =
+  | Date : Date.t t
+  | From : Mailbox.t list t
+  | Sender : Mailbox.t t
+  | ReplyTo : Address.t list t
+  | To : Address.t list t
+  | Cc : Address.t list t
+  | Bcc : Address.t list t
+  | Subject : Unstructured.t t
+  | MessageID : MessageID.t t
+  | InReplyTo : phrase_or_message_id list t
+  | References : phrase_or_message_id list t
+  | Comments : Unstructured.t t
+  | Keywords : phrase list t
+  | Field : Field_name.t -> Unstructured.t t
+  | Content : Content.t t
+  | Resent : Resent.t t
+  | Trace : Trace.t t
+
+type 'a v =
+  | Date : Date.t v
+  | Mailboxes : Mailbox.t list v
+  | Mailbox : Mailbox.t v
+  | Addresses : Address.t list v
+  | MessageID : Rfc822.nonsense Rfc822.msg_id v
+  | Phrase_or_message_id : phrase_or_message_id list v
+  | Unstructured : Unstructured.t v
+  | Phrases : phrase list v
+  | Content : Content.t v
+  | Resent : Resent.t v
+  | Trace : Trace.t v
```

Comment:
sorry for jumping in here, but I've a question: what is the relationship between `t` and `v`? both seem to be GADTs that carry name-value relationships..

Reply:
`'a t` is a field-name and `'a v` is type of values possible. I'm not clear about the design on them but, in my case, multiple field names can have the same kind of value (like `Cc` and `Bcc`). Then, an approach available in `Hd` module is to describe what we expect like: `Field "X-Received"` with `Trace` kind of value.

Reply:
ah, thanks for the quick reply!

Reply:
[This type](https://github.com/mirage/mrmime/blob/master/lib/hd.mli#L14) should be equal to `'a Field.v`.